lib/svc/method/net-virtual

```diff
@@ -0,0 +1,563 @@
+#!/usr/bin/ksh93
+#
+# CDDL HEADER START
+#
+# The contents of this file are subject to the terms of the
+# Common Development and Distribution License (the "License").
+# You may not use this file except in compliance with the License.
+#
+# You can obtain a copy of the license at usr/src/OPENSOLARIS.LICENSE
+# or http://www.opensolaris.org/os/licensing.
+# See the License for the specific language governing permissions
+# and limitations under the License.
+#
+# When distributing Covered Code, include this CDDL HEADER in each
+# file and include the License file at usr/src/OPENSOLARIS.LICENSE.
+# If applicable, add the following below this CDDL HEADER, with the
+# fields enclosed by brackets "[]" replaced with your own identifying
+# information: Portions Copyright [yyyy] [name of copyright owner]
+#
+# CDDL HEADER END
+#
+#
+# Copyright (c) 1984, 1986, 1987, 1988, 1989 AT&T. All rights reserved.
+# Copyright (c) 1999, 2010, Oracle and/or its affiliates. All rights reserved.
+# Copyright 2017 Joyent, Inc.
+# Copyright 2017 Erigones, s. r. o.
+#
+
+#
+# This script is responsible for creating VNICs (dladm create-vnic) based on
+# configuration in /usbkey/config or provided via boot parameters. It can also
+# create (replace) the overlay_rules.json file. Overlays are created only
+# if a VNIC is configured to be over an overlay. Etherstubs are created in
```

Comment:
.. if and underlying link of a VNIC is an overlay (in other words, overlays are created only if they are needed).